Source/WebCore/dom/TrustedTypePolicy.cpp

```diff
@@ -50,65 +51,68 @@ TrustedTypePolicy::TrustedTypePolicy(const String& name, const TrustedTypePolicy
 
-            return TrustedScript::create(contents);
-        }
-        if (callbackResult.type() == CallbackResultType::ExceptionThrown)
-            return Exception { ExceptionCode::ExistingExceptionError };
-    }
-
-    return Exception {
-        ExceptionCode::TypeError,
-        makeString("Policy "_s, m_name,
-        "'s TrustedTypePolicyOptions did not specify a 'createScript' member."_s)
-    };
+    return TrustedScript::create(policyValue.releaseReturnValue());
 }
 
 ExceptionOr<Ref<TrustedScriptURL>> TrustedTypePolicy::createScriptURL(const String& input, FixedVector<JSC::Strong<JSC::Unknown>>&& arguments)
 {
-    if (m_createScriptURLCallback) {
-        auto callbackResult = m_createScriptURLCallback->handleEvent(input, WTFMove(arguments));
+    auto policyValue = getPolicyValue(TrustedType::TrustedScriptURL, input, WTFMove(arguments));
+
+    if (policyValue.hasException())
+        return policyValue.releaseException();
 
-        if (callbackResult.type() == CallbackResultType::Success) {
-            auto contents = callbackResult.releaseReturnValue();
+    return TrustedScriptURL::create(policyValue.releaseReturnValue());
+}
+
+// https://w3c.github.io/trusted-types/dist/spec/#get-trusted-type-policy-value-algorithm
+ExceptionOr<String> TrustedTypePolicy::getPolicyValue(const TrustedType trustedTypeName, const String& input, FixedVector<JSC::Strong<JSC::Unknown>>&& arguments, bool throwIfMissing)
+{
+    CallbackResult<String> policyValue(CallbackResultType::UnableToExecute);
+    if (trustedTypeName == TrustedType::TrustedHTML) {
+        if (m_createHTMLCallback)
+            policyValue = m_createHTMLCallback->handleEvent(input, WTFMove(arguments));
+    } else if (trustedTypeName == TrustedType::TrustedScript) {
+        if (m_createScriptCallback)
+            policyValue = m_createScriptCallback->handleEvent(input, WTFMove(arguments));
+    } else if (trustedTypeName == TrustedType::TrustedScriptURL) {
+        if (m_createScriptURLCallback)
+            policyValue = m_createScriptURLCallback->handleEvent(input, WTFMove(arguments));
+    } else {
+        ASSERT_NOT_REACHED();
+        return Exception {
+            ExceptionCode::TypeError
+        };
+    }
 
-            return TrustedScriptURL::create(contents);
-        }
-        if (callbackResult.type() == CallbackResultType::ExceptionThrown)
-            return Exception { ExceptionCode::ExistingExceptionError };
+    if (policyValue.type() == CallbackResultType::Success)
+        return policyValue.releaseReturnValue();
+    if (policyValue.type() == CallbackResultType::ExceptionThrown)
+        return Exception { ExceptionCode::ExistingExceptionError };
+
+    if (throwIfMissing) {
+        return Exception {
+            ExceptionCode::TypeError,
+            makeString("Policy "_s, m_name,
+            "'s TrustedTypePolicyOptions did not specify a create'"_s, trustedTypeToString(trustedTypeName).substring(7), "' member."_s)
```

Comment:
This is not the traditional way to indent.

Calling `substring(7)` on the result of `trustedTypeToString` is an ugly way to handle it; we can do better. Maybe more than one function that returns a string for a `TrustedType`.